nestkernel/event.h

```diff
@@ -827,13 +843,20 @@ class GapJEvent : public SecondaryEvent
   static void
   set_syn_id( const synindex synid )
   {
-    synid_ = synid;
+    supported_syn_ids_.push_back( synid );
+  }
+
+  void
+  add_syn_id( const synindex synid )
+  {
+    supported_syn_ids_.push_back( synid );
   }
 
-  synindex
-  get_syn_id() const
+  bool
+  supports_syn_id( const synindex synid ) const
   {
-    return synid_;
+    return std::find( supported_syn_ids_.begin(), supported_syn_ids_.end(), synid )
+      != supported_syn_ids_.end();
```

Comment:
I'd put the entire expression in parentheses to make clear that the two line form one expression.